tools/challenge-md-parser/fix-challenges.js

```diff
@@ -0,0 +1,367 @@
+
+      file.data = { ...file.data, frontmatter };
+    }
+  }
+}
+
+function testsToData() {
+  return (tree, file) => {
+    visit(tree, 'code', visitor);
+
+    function visitor(node) {
+      const { lang, value } = node;
+      if (lang === 'yml') {
+        file.data = {
+          ...file.data,
+          tests: value
+        };
+      }
+    }
+  };
+}
+
+function textToData(sectionIds) {
+  return (tree, file) => {
+    let indexId = 0;
+    let currentSection = sectionIds[indexId];
+    let inSection = false;
+    let nodes = [];
+    let findSection;
+    const visitor = (node, index, parent) => {
+      if (!parent) {
+        return visit.CONTINUE;
+      }
+
+      if (node.type === 'heading') {
+        if (inSection) {
+          findSection = new RegExp(`^<section id=('|")${currentSection}\\1>`);
+          file.data = {
+            ...file.data,
+            [currentSection]: new stringify.Compiler(
+              { type: 'root', children: nodes },
+              file
+            )
+              .compile()
+              .trim()
+              .replace(findSection, '')
+              .replace(/<\/section>$/, '')
+              .trim()
+          };
+          nodes = [];
+          indexId++;
+          if (indexId < sectionIds.length) {
+            currentSection = sectionIds[indexId];
+          } else {
+            return visit.EXIT;
+          }
+        }
+        inSection = true;
+      } else if (inSection) {
+        nodes.push(node);
+      }
+
+      return visit.SKIP;
+    };
+    visit(tree, visitor);
+  };
+}
+
+function createTranslateText(target, source = 'english') {
+  const projectId = process.env.GOOGLE_CLOUD_PROJECT_ID;
+  if (!projectId) {
+    return async text => text;
+  }
+  const languageCodes = {
+    arabic: 'ar',
+    chinese: 'zh',
+    english: 'en',
+    portuguese: 'pt',
+    russian: 'ru',
+    spanish: 'es'
+  };
+  const from = languageCodes[source];
+  const to = languageCodes[target];
+  return async text => {
+    if (!text) {
+      return text;
+    }
+    try {
+      const translate = new Translate({ projectId });
+      const result = await translate.translate(text, { from, to });
+      const translations = result[0];
+      return translations;
+    } catch (err) {
+      // console.error(err);
+      return text;
+    }
+  };
+}
+
+async function processTests(tests, referenceTests, translateText) {
+  const testsObject = YAML.load(referenceTests);
+  if (
+    !testsObject.tests ||
+    testsObject.tests.length === 0 ||
+    !testsObject.tests[0].text
+  ) {
+    return referenceTests;
+  }
+  const newTests = await Promise.all(
+    testsObject.tests.map(async test => {
+      const text = await translateText(test.text);
+      return { ...test, text };
+    })
+  );
+  const testStrings = newTests
+    .map(
+      ({ text, testString }) =>
+        `  - text: ${dumpToYamlString(text)}    testString: ${dumpToYamlString(
+          testString
+        )}`
+    )
```

Comment:
Doesn't this function wipe already translated tests in favour of machine translation?